codec-http2/src/test/java/io/netty/handler/codec/http2/Http2MultiplexCodecTest.java

```diff
@@ -167,17 +181,69 @@ public void channelReadShouldRespectAutoRead() {
         verifyFramesMultiplexedToCorrectChannel(streamId, inboundHandler, 2);
     }
 
+    /**
+     * A child channel for a HTTP/2 stream in IDLE state (that is no headers sent or received),
+     * should not emit a RST_STREAM frame on close, as this is a connection error of type protocol error.
+     */
     @Test
-    public void streamClosedShouldFireChannelInactive() {
+    public void idleOutboundStreamShouldNotWriteResetFrameOnClose() {
+        childChannelInitializer.handler = new LastInboundHandler();
+
+        Http2StreamChannelBootstrap b = new Http2StreamChannelBootstrap();
+        b.parentChannel(parentChannel).handler(childChannelInitializer);
+        Channel childChannel = b.connect().channel();
+        assertTrue(childChannel.isActive());
+
+        childChannel.close();
+        parentChannel.runPendingTasks();
+
+        assertFalse(childChannel.isOpen());
+        assertFalse(childChannel.isActive());
+        assertNull(parentChannel.readOutbound());
+    }
+
+    @Test
+    public void outboundStreamShouldWriteResetFrameOnClose_headersSent() {
+        childChannelInitializer.handler = new ChannelInboundHandlerAdapter() {
+            @Override
+            public void channelActive(ChannelHandlerContext ctx) throws Exception {
+                ctx.writeAndFlush(new DefaultHttp2HeadersFrame(new DefaultHttp2Headers()));
+                ctx.fireChannelActive();
+            }
+        };
+
+        Http2StreamChannelBootstrap b = new Http2StreamChannelBootstrap();
+        b.parentChannel(parentChannel).handler(childChannelInitializer);
+        Channel childChannel = b.connect().channel();
+        assertTrue(childChannel.isActive());
+
+        parentChannel.flush();
```

Comment:
actually since you are doing a `writeAndFlush` in the channel active ... is this flush necessary?